server/src/main/java/org/opensearch/search/aggregations/bucket/terms/TermsAggregator.java

```diff
@@ -195,6 +196,29 @@ public boolean equals(Object obj) {
         }
     }
 
+    // BucketCountThresholds type that throws an exception when shardMinDocCount and shardSize are accessed. This is used for
+    // deserialization on the coordinator during reduce as shardMinDocCount and shardSize should not be accessed this way on the
+    // coordinator.
+    public static class CoordinatorBucketCountThresholds extends BucketCountThresholds {
+
+        public CoordinatorBucketCountThresholds(long minDocCount, long shardMinDocCount, int requiredSize, int shardSize) {
+            super(minDocCount, shardMinDocCount, requiredSize, shardSize);
+        }
+
+        @Override
+        public long getShardMinDocCount() {
+            throw new AggregationExecutionException("shard_min_doc_count should not be accessed");
+        }
+
+        @Override
+        public int getShardSize() {
+            if (shardSize < 0) {
+                throw new AggregationExecutionException("Invalid shard_size accessed");
+            }
+            return shardSize;
+        }
+    }
```

Comment:
This will still be called on the coordinator whenever `InternalTerms::reduce` calls `create` with the reduced buckets because that will pass the same `CoordinatorBucketCountThresholds` object, which is in turned used to set `shardSize` in the `InternalMappedTerms` constructor.

Reply:
Thanks for making the change to keep `shardSize` member for `InternalTerms`